File: Vanilla/sources/zdoc/media/lua/shared/Library/ZombieReanimateState.lua
---@class ZombieReanimateState : zombie.ai.states.ZombieReanimateState
---@field private _instance ZombieReanimateState
ZombieReanimateState = {}

---@public
---@param arg0 IsoGameCharacter
---@param arg1 AnimEvent
---@return void
function ZombieReanimateState:animEvent(arg0, arg1) end

---@public
---@param arg0 IsoGameCharacter
---@return void
function ZombieReanimateState:execute(arg0) end

---@public
---@param arg0 IsoGameCharacter
---@return void
function ZombieReanimateState:enter(arg0) end

---@public
---@param arg0 IsoGameCharacter
---@return void
function ZombieReanimateState:exit(arg0) end

---@public
---@return ZombieReanimateState
function ZombieReanimateState:instance() end

File: Vanilla/sources/zomboid/zombie/ai/states/ZombieReanimateState.java
package zombie.ai.states;

import zombie.ai.State;
import zombie.characters.IsoGameCharacter;
import zombie.characters.IsoZombie;
import zombie.core.skinnedmodel.advancedanimation.AnimEvent;

public final class ZombieReanimateState extends State {
   private static final ZombieReanimateState _instance = new ZombieReanimateState();

   public static ZombieReanimateState instance() {
      return _instance;
   }

   public void enter(IsoGameCharacter var1) {
      IsoZombie var2 = (IsoZombie)var1;
      var2.clearVariable("ReanimateAnim");
   }

   public void execute(IsoGameCharacter var1) {
   }

   public void exit(IsoGameCharacter var1) {
      IsoZombie var2 = (IsoZombie)var1;
      var2.clearVariable("ReanimateAnim");
   }

   public void animEvent(IsoGameCharacter var1, AnimEvent var2) {
      IsoZombie var3 = (IsoZombie)var1;
      if (var2.m_EventName.equalsIgnoreCase("FallOnFront")) {
         var1.setFallOnFront(Boolean.parseBoolean(var2.m_ParameterValue));
      }

      if (var2.m_EventName.equalsIgnoreCase("ReanimateAnimFinishing")) {
         var3.setReanimate(false);
         var3.setFallOnFront(true);
      }

      if (var2.m_EventName.equalsIgnoreCase("FallOnFront")) {
         var1.setFallOnFront(Boolean.parseBoolean(var2.m_ParameterValue));
      }

   }
}
